app/admin/friendship.rb

```diff
@@ -0,0 +1,37 @@
+ActiveAdmin.register Approval, as: "Friendship" do
+  actions :all, except: [:new, :create]
+
+  controller do
+    def scoped_collection
+      end_of_association_chain.where id: Approval.find_by_sql(
+        <<-SQL
+          select distinct on (token) *
```

Comment:
A little bit tricky way to get rid of duplicates.
We need paired approvals to be created one by one.
That's where [this change](https://github.com/earlymarket/Coposition/pull/696/files#diff-3b31d40a5ea5deb70f57c1f215ad9186R39) came from.